src/vm.nit

```diff
@@ -377,6 +397,84 @@ class VirtualMachine super NaiveInterpreter
 
 		return i != initialization_value
 	end
+
+	redef fun stack_trace: String
+	do
+		var b = new FlatBuffer
+		b.append(",---- Stack trace -- - -  -\n")
+		for f in numbered_frames do
+			b.append("| {f.mpropdef} ({f.current_node.location})\n")
+		end
+		b.append("`------------------- - -  -")
+		return b.to_s
+	end
+
+	redef fun current_node: nullable ANode
+	do
+		if numbered_frames.is_empty then return null
+		return current_frame.current_node
+	end
+
+	redef fun current_receiver_class: MClassType
+	do
+		return current_frame.arguments.first.mtype.as(MClassType)
+	end
+
+	redef fun is_escaping: Bool do return nreturnmark != null or escapemark != null
+
+	# Evaluate `n` as an expression in the current context.
+	# Return the value of the expression.
+	# If `n` cannot be evaluated, then aborts.
+	redef fun expr(n: AExpr): nullable Instance
+	do
+		var frame = self.current_frame
+		var old = frame.current_node
+		frame.current_node = n
+
+		var i = n.expr(self)
+		if i == null and not self.is_escaping then
+			n.debug("inconsitance: no value and not escaping.")
+		end
+		var implicit_cast_to = n.implicit_cast_to
+		if implicit_cast_to != null then
+			var mtype = self.unanchor_type(implicit_cast_to)
+			if not self.is_subtype(i.mtype, mtype) then n.fatal(self, "Cast failed. Expected `{implicit_cast_to}`, got `{i.mtype}`")
```

Comment:
Here it can be evaluated but the method aborts because of typing?

Yeah, there is something I don't understand this the comment `Is cannot be evaluated, the aborts`.